WordPress/gutenberg
# Border Support: Add support for custom border units

## Description
- Allows configuration of available CSS units for border width and radius support. 
- Small CSS tweak made to border style component's `legend` element so that it doesn't stretch the border width UnitControl's height.

## How has this been tested?
Manually.

1. Checkout this PR
2. Create a new post and add a group block
3. Select the group block, expand border support panel in sidebar, confirm width and radius fields offer the same units as the padding controls under "Spacing"
4. Within your theme.json under `settings.spacing` change `units` as follows: `"units": [ "px", "em" ]`.
5. Reload the editor and confirm that the width and radius fields now only offer `px` and `em` units.
6. Switch to the site editor, expand the global styles panel, click the by block type tab, then expand the group panel.
7. Confirm the units on the border width and radius fields are correct.

## Screenshots 

<img width="279" alt="Screen Shot 2021-07-09 at 11 48 33 am" src="https://user-images.githubusercontent.com/60436221/125011566-0fdbec80-e0ac-11eb-8fa3-a184cf0d9459.png">

## Types of changes
Bug fix.

## Checklist:
